File: app/src/main/java/com/di/penopllast/vklikesremover/di/module/ContextModule.java
package com.di.penopllast.vklikesremover.di.module;

import android.content.Context;

import com.di.penopllast.vklikesremover.application.DislikerApp;

import dagger.Module;
import dagger.Provides;

@Module
public class ContextModule {

    @Provides
    Context provideContext(DislikerApp app) {
        return app.getApplicationContext();
    }

}

File: app/src/main/java/com/di/penopllast/vklikesremover/di/module/ContextModule.kt
package com.di.penopllast.vklikesremover.di.module

import android.content.Context

import com.di.penopllast.vklikesremover.application.DislikerApp

import dagger.Module
import dagger.Provides

@Module
class ContextModule {

    @Provides
    internal fun provideContext(app: DislikerApp): Context {
        return app.applicationContext
    }

}
